registry/client/repository.go

```diff
@@ -131,7 +139,7 @@ func (r *registry) Repositories(ctx context.Context, entries []string, last stri
 }
 
 // NewRepository creates a new Repository for the given repository name and base URL.
-func NewRepository(name reference.Named, baseURL string, transport http.RoundTripper) (distribution.Repository, error) {
+func NewRepository(name reference.Named, baseURL string, transport http.RoundTripper, jar http.CookieJar) (distribution.Repository, error) {
```

Comment:
I'm curious why we add a `http.CookieJar` as a parameter on `NewRepository` without adding an instance to it in `proxyingRegistry.Respositories`.